tests/python/unittest/test_ndarray.py

```diff
@@ -711,9 +711,8 @@ def get_values(ensure_unique):
                  k=dat_size*dat_size*dat_size*dat_size, is_ascend=False)
     assert_almost_equal(nd_ret_argsort, gt)
 
-    # test topk with a big shape
-    a = mx.nd.arange(0, 54686454, step=1, repeat=1)
-    assert_almost_equal(a.topk(k=54686454).asnumpy(), a.asnumpy()[::-1])
+    a = mx.nd.arange(0, 1024, step=1, repeat=1)
+    assert_almost_equal(a.topk(k=1024).asnumpy(), a.asnumpy()[::-1])
```

Comment:
As stated before, leave this test as is. You're defeating the purpose of this test with this modification.

Reply:
I have discussed at length that is not ok to reserve a 4gb ndarray. The author that introduced this test should have done a cpp unit test for the specific regression that he targetted. This is blocking raspberry pi validation. We have customers that need to run the core unit tests on device, as well as we do. Please don't block our work of bringing mxnet to resource constraint devices.

Reply:
But your patch effectively defeats the purpose of this test. As you said, it was lenghtly discussed and you were always told to write a cpp test and to not disable / change this test. So either write a cpp test and remove this one or leave it alone.

Reply:
I timeboxed the time to write a CPP test for this, it's not trivial, and it's out of the scope of my work.

For me, having this test adjusted to a reasonable parameter is a necessary condition to have ARM support offered. I would suggest to file a ticket to do this task in the future, move it to nightly, or write it yourself. It's out of the scope of ARM support to test for an overflow in topk in the tests for ndarray which have to run on all platforms.

Reply:
https://github.com/apache/incubator-mxnet/pull/8398#discussion_r185568286

Reply:
since mshadow::range has been fixed to use double instead of float here: https://github.com/dmlc/mshadow/pull/301 , I think it is fine to add the CPP test in the backlog. The risk is that if mshadow::range were to rollback to using float and then mxnet is updated to use that mshadow and some op in mxnet uses mshadow::range this would cause an uncaught bug. I see mshadow::range being used in two ops currently :: bounding_box and grid_generator. This may also change soon since mshadow is being deprecated.  I think the risk is low to justify adding this to backlog.

Reply:
I agree this PR should not be blocked by this test. What about moving adding this to nightly tests? We can create a JIRA/github issue to track it (and make sure it's added back)

Reply:
Tracked at https://github.com/apache/incubator-mxnet/issues/11297